### ISSUE TITLE
ESP8266 aREST Local Access ?

### ISSUE THREAD

**Issue by nbibin001** [marcoschwartz/aREST#127]
Thank you for this aREST library and detailed examples. I was playing with it for my ESP8266 projects. I was curious to know more about the ESP8266_cloud_and_local option that you added recently. So I was able to control the ESP through cloud and also locally using the fixed IP address. But would like to have the ESPs setup up with dynamic IP addresses, (so the IP is not fixed) and accessible via aREST locally (local wifi, no via internet). Will that be possible? I would really appreciate your inputs. 

Thanks and Regards
Bibin

**Comment by saliehendricks** [comment]
Hi Bibin

That is possible. I have my ESPs all use dynamic IP addresses. You would need to know the IPs to be able to connect to them though. In my case I broadcast the IPs using multicast broadcast after wifi is connected. I have another machine that listens for broadcast packets and keeps track of the IPs and exposes them locally via a little webserver app I wrote. I am quite familiar with C# so I used that to do my broadcast receiver and web app code. Your server can be something that you are familiar with.

On the ESP
```
WifiUDP Udp;
IPAddress ipMulti (192,168,10,255); //<--- note the .10 is specific to ur local network range
unsigned int localPort = 1244;      // local port to listen on for udp
...

I call this in setup:

void multicast_config() //When ESP8266 starts, Broadcast my presence 
{
	IPAddress localIp = WiFi.localIP();
	IPAddress broadCast = (localIp[0], localIp[1],localIp[2] ,255);
	Udp.beginPacket(ipMulti, broadCast);    //localPort
	Udp.write(udpConfig.c_str());
	Udp.endPacket();
 
} //- See more at: http://www.esp8266.com/viewtopic.php?p=16377#sthash.gugFiDFN.dpuf
```

**Comment by nbibin001** [comment]
Hello, thanks for the detailed explanation with example. I guess this will do the job for me. One question though is using the "WifiUDP" package I should be able to receive the multicast packets sent by the ESP's after connecting to wifi right ?  I am planning to have both the controlling and controlled devices based on ESP's wifi and aREST. So if the device can broadcast its dynamic IP along with its device id or some other unique information which can be received by other controller ESP and form a IP-Device ID table kind of a deal.

**Comment by saliehendricks** [comment]
Yes that is correct. I send a bit of json - the string above called udpConfig looks something like: 
udpConfig = "bar|{\"nodeId\":\""+ nodeId + "\"}"; You will notice I dont even bother sending the IP cos the receiver if that broadcast will be able to know source IP from the packet info. You can add more data but there is a UDP packet limit. Its pretty small so keep it limited to say 300 chars. I think that buffer size is 4Kb   if I recall.
